src/NServiceBus.Core/Recoverability/RecoverabilityComponent.cs

```diff
@@ -124,7 +124,8 @@ RecoverabilityExecutorFactory CreateRecoverabilityExecutorFactory(IBuilder build
                 immediateRetriesAvailable,
                 delayedRetriesAvailable,
                 MessageRetryNotification,
-                MessageFaultedNotification);
+                MessageFaultedNotification,
+                builder.Build<IDispatchMessages>());
```

Comment:
it looks like the dispatcher is build several times (but a bit later since we're only passing factories), given that the builder is created here anyway, could that be reused?

Reply:
I didn't bother because it is a singleton anyway but yes we could extract it into a local